rust/main/hyperlane-base/src/contract_sync/mod.rs

```diff
@@ -108,6 +109,11 @@ where
                     .await;
             }
         }
+
+        // Although the above loop should never end (unless by panicking),
+        // we put log here to make sure that we see when this method returns normally.
```

Comment:
so this is if we ever change this function such that the loop ever exits, then we have this log?

Reply:
This log is added out of desperation why task stopped making any progress. Presumably, compiler should optimise it out, but just in case there is a weird exit from the loop (which is not a panic), we may see this log.